components/renku_data_services/namespace/orm.py

```diff
@@ -74,21 +75,43 @@ class NamespaceORM(BaseORM):
     )
     user: Mapped[UserORM | None] = relationship(lazy="joined", init=False, repr=False, viewonly=True)
 
+    @hybridproperty
+    def created_by(self) -> str:
+        """User that created this namespace."""
+        if self.group is not None:
+            return self.group.created_by
+        elif self.user_id is not None:
+            return self.user_id
+
+        raise errors.ProgrammingError(
+            message=f"Found a namespace {self.slug} that has no group or user associated with it."
+        )
+
+    @hybridproperty
+    def creation_date(self) -> datetime | None:
+        """When this namespace was created."""
+        if self.group is not None:
+            return self.group.creation_date
+        return None
```

Comment:
I wonder if we can get this from Keycloak somehow. Maybe we should store it?

Reply:
there's a creationTimestamp in https://www.keycloak.org/docs-api/latest/rest-api/index.html#UserRepresentation so we can definitely get it in the full user sync. There doesn't seem to be documentation of the user events payload https://www.keycloak.org/docs-api/latest/rest-api/index.html#EventRepresentation at least not that I could find, though if there isn't a creation time in there, we could use the event timestamp of user register events.